pkg/kubelet/container/helpers.go

```diff
@@ -104,3 +108,47 @@ func ExpandContainerCommandAndArgs(container *api.Container, envs []EnvVar) (com
 
 	return command, args
 }
+
+// Create an event recorder to record object's event except implicitly required container's, like infra container.
+func FilterEventRecorder(recorder record.EventRecorder) record.EventRecorder {
+	return &innerEventRecorder{
+		recorder: recorder,
+	}
+}
+
+type innerEventRecorder struct {
+	recorder record.EventRecorder
+}
+
+func (irecorder *innerEventRecorder) Event(object runtime.Object, reason, message string) {
+	if object == nil {
```

Comment:
`dockertools/manager.go` is full of if statement like this:
```
if ref != nil {
     dm.recorder.Eventf(ref, "Failed", "Failed to create docker container with error: %v", err)
}
```
Since you check whether the object is nil, can we also remove all the `if ref != nil` code?